src/Lunapark/Error.purs

```diff
@@ -1,123 +1,24 @@
-  "invalid selector" → Right InvalidSelector
-  "invalid session id" → Right InvalidSessionId
-  "javascript error" → Right JavaScriptError
-  "move target out of bounds" → Right MoveTargetOutOfBounds
-  "no such alert" → Right NoSuchAlert
-  "no such cookie" → Right NoSuchCookie
-  "no such element" → Right NoSuchElement
-  "no such frame" → Right NoSuchFrame
-  "no such window" → Right NoSuchWindow
-  "script timeout" → Right ScriptTimeout
-  "session not created" → Right SessionNotCreated
-  "stale element reference" → Right StaleElementReference
-  "timeout" → Right Timeout
-  "unable to set cookie" → Right UnableToSetCookie
-  "unable to capture screen" → Right UnableToCaptureScreen
-  "unexpected alert open" → Right UnexpectedAlertOpen
-  "unknown command" → Right UnknownCommand
-  "unknown error" → Right UnknownError
-  "unknown method" → Right UnknownMethod
-  "unsupported operation" → Right UnsupportedOperation
-  s → Left s
-
-toStringCode ∷ ErrorType → String
-toStringCode = case _ of
-  ElementClickIntercepted → "element click intercepted"
-  ElementNotSelectable → "element not selectable"
-  ElementNotInteractable → "element not interactable"
-  InsecureCertificate → "insecure certificate"
-  InvalidArgument → "invalid argument"
-  InvalidCookieDomain → "invalid cookie domain"
-  InvalidCoordinates → "invalid coordinates"
-  InvalidElementState → "invalid element state"
-  InvalidSelector → "invalid selector"
-  InvalidSessionId → "invalid session id"
-  JavaScriptError → "javascript error"
-  MoveTargetOutOfBounds → "move target out of bounds"
-  NoSuchAlert → "no such alert"
-  NoSuchCookie → "no such cookie"
-  NoSuchElement → "no such element"
-  NoSuchFrame → "no such frame"
-  NoSuchWindow → "no such window"
-  ScriptTimeout → "script timeout"
-  SessionNotCreated → "session not created"
-  StaleElementReference → "stale element reference"
-  Timeout → "timeout"
-  UnableToSetCookie → "unable to set cookie"
-  UnableToCaptureScreen → "unable to capture screen"
-  UnexpectedAlertOpen → "unexpected alert open"
-  UnknownCommand → "unknown command"
-  UnknownError → "unknown error"
-  UnknownMethod → "unknown method"
-  UnsupportedOperation → "unsupported operation"
-
-type Error =
-  { error ∷ ErrorType
-  , message ∷ String
-  , stacktrace ∷ String
-  }
-
-fromJson ∷ J.Json → Error
-fromJson js = either unknownError identity do
-  obj ← J.decodeJson js
-  value ← obj J..: "value"
-  error ← fromStringCode =<< value J..: "error"
-  message ← value J..: "message"
-  stacktrace ← value J..: "stacktrace"
-  pure { error, message, stacktrace }
-
-unknownError ∷ String → Error
-unknownError message =
-  { error: UnknownError
-  , message
-  , stacktrace: ""
-  }
+import Affjax as N
+import Lunapark.WebdriverError as LWE
+import Data.Argonaut.Decode (JsonDecodeError, printJsonDecodeError) as J
+
+data Error
+  = JsonDecodeError J.JsonDecodeError
+  | WebdriverError LWE.WebdriverError
+  | AffjaxError N.Error
+  | UserError String
```

Comment:
This one is actually `LunaparkError` not `UserError` as it's internal thing for lunapark

Reply:
@cryogenian

how about rename `UserError` to `InternalError` (since it is only in lib)

and rename `LunaparkF` to `BaseActionF`?

 I think there is too much `Lunapark` in lunapark

Reply:
or maybe LunaparkF to ActionF and ActionF to CompoundActionF?

Reply:
Well it's not internal, right? More like it's compound or domain level, what about `CompoundError`? 

`LunaparkF -> ActionF` and `ActionF -> CompoundActionF` break webdriver naming scheme, because `ActionF` is actually about actions in webdriver (they are encoded as `action` field and discussed in `action` section of docs. I'd agree that `LunaparkActionEffect -> ActionEffect` is OK, but, yeah, `ActionEffect` sounds a bit too generic to mee, although it's definitely acceptable.

Reply:
1. re errors

I think I have found the best variant https://github.com/slamdata/purescript-lunapark/pull/13/commits/e7428d973c198bbb89d316e1ec2418ad9fdd95c2

2. re renaming

I have finally understood the ActionF

it transforms

```
my = do
  ActionF.sendKeys "a"
  ActionF.pause
  LunaparkF.foo
  ActionF.sendKeys "a"
  ActionF.pause
```

to

```
my = do
  LunaparkF.performActions $ Object.fromHomogenous
    { "id1": Key [keyDown "a", keyUp "a", pause]
    }
  LunaparkF.foo
  LunaparkF.performActions $ Object.fromHomogenous
    { "id1": Key [keyDown "a", keyUp "a", pause]
    }
```

spec https://github.com/jlipps/simple-wd-spec#perform-actions

but also supports deprecated json wire

wow this complicated

_________

3. renaming is done in e60fd79

Reply:
Yep, that's how it works. We had problems where ff driver and chromedriver were supporting only subsets of actual webdriver API.